Qt5 dependency on Kinetic - Cannot build with Qt4

rviz will not build with Qt4 on branch **kinetic-devel** : 
```
.../rviz/src/rviz/visualization_frame.cpp:1253:66: error: ‘FindDirectChildrenOnly’ is not a member of ‘Qt’
   QList<QTabBar *> tab_bars = findChildren<QTabBar *>(QString(), Qt::FindDirectChildrenOnly);
```
[Merged PR #1168](https://github.com/ros-visualization/rviz/pull/1168) relies on `Qt::FindDirectChildrenOnly` to get a list of all `QTabBar` children of `VisualizationFrame`, which is a flag to `QObject::findChild()` that was introduced in Qt5.

See `visualization_frame.cpp`, line 1253 :  
```cpp
QList<QTabBar *> tab_bars = findChildren<QTabBar *>(QString(), Qt::FindDirectChildrenOnly);
```
Qt4 does not seem to provide an "easy" equivalent of this flag. The subject of reproducing its behaviour has been covered [here](https://stackoverflow.com/questions/8740278/is-there-a-simple-way-of-limiting-qobjectfindchild-to-direct-children-only), but proposed solutions rely on the children's object names, which I am not sure will be reliable in the context of rviz.

### Your environment
* OS Version: e.g. Ubuntu 16.04
* ROS Distro: Kinetic
* RViz, Qt, OGRE, OpenGl version as printed by rviz: 
    ```
    no build, Qt 4.8.7
    ```

--- a/src/rviz/visualization_frame.cpp
+++ b/src/rviz/visualization_frame.cpp
@@ -1250,10 +1250,13 @@ void VisualizationFrame::onHelpAbout()
 
 void VisualizationFrame::onDockPanelChange()
 {
-  QList<QTabBar *> tab_bars = findChildren<QTabBar *>(QString(), Qt::FindDirectChildrenOnly);
+  QList<QTabBar *> tab_bars = findChildren<QTabBar *>(QString());
   for ( QList<QTabBar *>::iterator it = tab_bars.begin(); it != tab_bars.end(); it++ )
   {
-    (*it)->setElideMode( Qt::ElideNone );
+	if((*it)->parent() == this)
+	{
+		(*it)->setElideMode( Qt::ElideNone );
+	}
   }
 }
 

--- a/src/rviz/visualization_frame.cpp
+++ b/src/rviz/visualization_frame.cpp
@@ -1250,10 +1250,17 @@ void VisualizationFrame::onHelpAbout()
 
 void VisualizationFrame::onDockPanelChange()
 {
+#if QT_VERSION >= QT_VERSION_CHECK(5, 0, 0)
+  QList<QTabBar *> tab_bars = findChildren<QTabBar *>(QString(), Qt::FindDirectChildrenOnly);
+#else
   QList<QTabBar *> tab_bars = findChildren<QTabBar *>(QString());
+#endif
+
   for ( QList<QTabBar *>::iterator it = tab_bars.begin(); it != tab_bars.end(); it++ )
   {
+#if QT_VERSION < QT_VERSION_CHECK(5, 0, 0)
 	if((*it)->parent() == this)
+#endif
 	{
 		(*it)->setElideMode( Qt::ElideNone );
 	}

